### ISSUE TITLE
Remove custom formatter

### ISSUE THREAD

**Issue by ioquatix** [ruby/error_highlight#15]
https://github.com/ruby/error_highlight/blob/9fbaa8ab7c09257320afb39ae4619251362291b1/lib/error_highlight/formatter.rb#L16

I believe it would be prudent at least for the first stable release of this gem, to remove the `ErrorHighlight#formatter` override.

I believe it will be very hard to use in practice and adds unnecessary public interface and complexity.

**Comment by eregon** [comment]
Agreed, ErrorHighlight is an internal gem, it should not expose any public method.
Let's design this kind of things [properly](https://bugs.ruby-lang.org/issues/18296) and not make them ErrorHighlight-specific.
